sql/migrations/0001_initial.sql

```diff
@@ -113,6 +113,15 @@ CREATE TABLE atxs
 CREATE INDEX atxs_by_smesher_by_epoch_desc ON atxs (smesher, epoch desc);
 CREATE INDEX atxs_by_epoch_by_pubkey ON atxs (epoch, smesher);
 
+CREATE TABLE vrf_nonces
```

Comment:
there is an option to store the last one in the identities table

Reply:
I decided on a dedicated table for VRF Nonces because of the following:

- most ATXs published by a node will not contain a VRF Nonce, it is only necessary that the initial ATX contains one
- if a node publishes a new VRF Nonce (lets say in epoch 50) it becomes effective the epoch after any messages published before epoch 51 use the initial VRF Nonce and any messages after epoch 50 will use the new one

> what do you think of adding vrf_nonces field in the atxs table?
> conceptually they should be a property of ATX, instead of having a separate life-cycle.

VRF nonces have their own life cycle, the majority of ATXs will not contain a VRF Nonce. A significant amount of smeshers will probably never update their initially published VRF Nonce. Only smeshers that decide to increase their PoST at some point after they join the network will start using a different VRF Nonce starting from the epoch they use the increased PoST.

> when you query, just pick the nonce in atxs table with the highest epoch.

That would be wrong, since it depends on which epoch the message was published in which VRF Nonce has to be queried from the table. Lets say Smesher A started with 2 `numUnits` of PoST size and smeshed for a few months, then in epoch 10 they decide to increase their PoST size to 4 `numUnits` and publish a new VRF Nonce that has to be used starting from epoch 11. Now a new Smesher B comes online and syncs with the network. To be able to verify messages published by Smesher A in epoch 1 - 10 they have to use the initial VRF Nonce, but starting from epoch 11 they have to use the updated value.

Reply:
After discussing the topic on more detail with @countvonzero I changed the implementation such that the VRF Nonce is now a (nullable) column in the atxs table and updated the code for persisting and fetching the nonce accordingly.

This allows in the future to more easily prune local state by just considering ATXs, and keep at least the newest ATX of an identity that contains a VRF Nonce instead of having to manage two tables.

Reply:
> This allows in the future to more easily prune local state by just considering ATXs, and keep at least the newest ATX of an identity that contains a VRF Nonce instead of having to manage two tables.

but this is irrelevant. if identity is cancelled it doesn't matter what is a previous vrf nonce